docs/GettingStarted.md

```diff
@@ -1936,6 +1941,91 @@ Then you must enable the request queuing feature in the integration handling the
+
+```ruby
+    Datadog.configure do |c|
+      c.use :rack, rum_injection_enabled: true, rum_injection_disabled_paths: ['/api', '/blog/**/**']
+    end
+```
+
+Review the Rack [documentation](#rack) for more configuration details.
+
+When the Rack RUM Injection Middlewarre is used in conjunction with the Ruby on Rails instrumentation, the middleware is automatically inserted into your application's rack middleware stack in the appropriate location. Here is an example configuration:
+
+
+```ruby
+    Datadog.configure do |c|
+      c.use :rack, rum_injection_enabled: true, rum_injection_disabled_paths: ['/api', '/blog/**/**']
+      c.use :rails
+    end
+```
+
+However, when using Rack as a standalone application or in conjunction with another web framework, such as Sinatra, you must manually insert the Rack RUM Injection Middleware at the appropriate location within the middleware stack, via `use Datadog::Contrib::Rack::RumInjection`. It should be the _last_ middleware in the stack, and come _after_ any compression middleware such as `Rack::Deflater`
+
+```ruby
+# config.ru example
+require 'ddtrace'
+
+Datadog.configure do |c|
+  c.use :rack, options
+end
+
+use Datadog::Contrib::Rack::TraceMiddleware
+use Rack::Deflater
+use Datadog::Contrib::Rack::RumInjection
+
+app = proc do |env|
+  [ 200, {'Content-Type' => 'text/plain'}, ['OK'] ]
+end
+
+run app
+```
+
+Last, ensure the [`browser-sdk`](https://docs.datadoghq.com/real_user_monitoring/installation/?tab=us) is setup correctly in your front-end web application.
+
+The RUM Injection middleware will insert an HTML comment into only those responses that are `Content-Type` `html` or `xhtml`, can be reasonably determined to be non-cached html responses at either browser or CDN level, are not streaming responses, and are not compressed or gzipped at time of injection.
+
+
+#### RUM Manual Injection
+
+For users that have a caching strategy for their HTML that leverages a CDN, VCL, or custom caching rules, the automatic RUM Injection may not be suitable for determining which html templates are not cached, and should have a trace-id injected.  In this case, we also provide a Manual Injection option so that users can configure which HTML templates specifically they should inject the trace-id into. The Manual Injection template helpers inject an HTML `<meta>` tag containing the `dd-trace-id` and `dd-trace-time`. This allows the [`browser-sdk`](https://docs.datadoghq.com/real_user_monitoring/installation/?tab=us) to connect frontend sessions to backend traces.  To modify a template, add the template helper to generate the RUM Injection meta tags (we recommend `<head>` section of your template, but it can be added anywhere)
+
+```
+  <head>
+    <%= ::Datadog::Contrib::Rack::RumInjection.inject_rum_data rescue "" %>
```

Comment:
I think we should move the Rails example first.
So the order of examples would be:
* Rails
* Other Rack apps
* Generic

As this would be the popularity order of these solutions.

Reply:
yup agreed, updated